plugins/browser-plugin-media-tracking/src/helperFunctions.ts

```diff
@@ -0,0 +1,102 @@
+import { DefaultEvents, EventGroups } from './eventGroups';
+import { DocumentEvent, MediaEvent, TextTrackEvent } from './mediaEvents';
+import { SnowplowMediaEvent } from './snowplowEvents';
+import { EventGroup, RecievedTrackingOptions, TextTrackObject, TrackingOptions } from './types';
+
+export function timeRangesToObjectArray(t: TimeRanges): { start: number; end: number }[] {
+  let out = [];
+  for (let i = 0; i < t.length; i++) {
+    out.push({ start: t.start(i), end: t.end(i) });
+  }
+  return out;
+}
+
+export function textTrackListToJson(textTrackList: TextTrackList): TextTrackObject[] {
+  let out: TextTrackObject[] = [];
+  for (let o of Object.keys(textTrackList)) {
+    let i = parseInt(o);
+    out.push({
+      label: textTrackList[i].label,
+      language: textTrackList[i].language,
+      kind: textTrackList[i].kind,
+      mode: textTrackList[i].mode,
+    });
+  }
+  return out;
+}
+
+export function isTypeTextTrackEvent(e: string): boolean {
+  let fields: string[] = Object.keys(TextTrackEvent);
```

Comment:
```suggestion
  const fields: string[] = Object.keys(TextTrackEvent);
```

I think you get the idea by now ;-)